### ISSUE TITLE
dtbo file not found when building with DT_FROM_BD_ENABLE = 0

### ISSUE THREAD

**Issue by rbroucquart** [MicroTCA-Tech-Lab/meta-techlab-utils#3]
When `DT_FROM_BD_ENABLE = 0` the per variant device tree overlays (dtbo) are not generated.

In append `fpga-manager-util_%.bbappend` the `do_install_prepend` always try to install dtbo, regardless of their creation.

Maybe it can be fixed by conditioning this install to  `DT_FROM_BD_ENABLE = 1` ?

Or should the dtbo file be provided elsewhere ?

Thanks !

**Comment by patrislav1** [comment]
this is a use case we kind of skipped till now, we either used both `PL_VARIANTS` and `DT_FROM_BD_ENABLE`, or none of them.

Do you want/need a `.dtbo` for the PL, or can you do without one?
Are you using techlab BSP for the PL or the FWK?

**Comment by rbroucquart** [comment]
Hello Patrick,

sure I understand, and I think the good point is that Yocto layers and code sharing allows expansion.

For full explanation, here's how I'm proceeding.
I have two FPGA applications, related but different.
They're based on FWK project, with the FWK BSP (and desyRDL axi arbiter, and no blockdesign).

I wanted to be able to select the application after boot. I used PL_VARIANTS for that.
Luckily, these two application can have the same device tree (mainly because the difference will be cover by a different dmap in ChimeraTK, and that the DT only contains a UIO then)
So I don't use a dtbo. (To be frank I tried once and failed, so moved on)

I go one step further with the PL_VARIANT feature, by having also PL_VARIANT for the next version. So I'm able to switch and roll back, with the OS still running and by only having to reload ChimeraTK app if register mapping changed.

I used to patch your recipe by adding a if condition  `DT_FROM_BD_ENABLE = 1 ` 
I'm in the process of cleaning/straightening  things, so I published this "issue"
One other way could be for me to overlay your layer, but that would be the same code for every other line of the function.

thanks again for your answer, and of course for the sharing of all this very good work.
